template/module/static/src/js/module_name.js

```diff
@@ -1,7 +1,20 @@
 /* Copyright <YEAR(S)> <AUTHOR(S)>
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
 
-(function ($) {
+odoo.define('module_name.object_name', function(require){
     "use strict";
-    // Script that will be loaded when document is ready
-})(jQuery);
+
+    var core = require('web.core');
+
+    core.bus.on('web_client_ready', null, function () {
+        // Script that will be loaded when document is ready
+    });
```

Comment:
This approach doesn't work for `website`. Instead, the proper way to determine that the document is ready seems to be via `ready()` in `web_editor.base`, which returns a Promise object. This means that for `website`, the example here should look more like:

```
var base = require('web_editor.base');
base.ready().done(function() {
    // Script that will be loaded when document is ready
});
```